src/core/clients/TV.ts

```diff
@@ -0,0 +1,226 @@
+    
+    if (item.is(HorizontalList)) {
+      continuation = item.continuations?.first()?.continuation;
+    } else if (item.is(HorizontalListContinuation)) {
+      continuation = item.continuation;
+    } else {
+      throw new InnertubeError(`No supported YTNode supplied. Type: ${item.type}`);
+    }
+    
+    if (!continuation) {
+      throw new InnertubeError('No continuation data available.');
+    }
+    
+    const data = await this.#actions.execute('/browse', {
+      client: client ?? 'TV',
+      continuation: continuation
+    });
+
+    const parser = Parser.parseResponse<IBrowseResponse>(data.data);
+    return parser.continuation_contents;
+  }
+  
+  // Interactions for TV (for default OAuth login)
+
+  /**
+   * Adds videos to a given playlist.
+   * @param playlist_id - The playlist ID.
+   * @param video_ids - An array of video IDs to add to the playlist.
+   * @param client - Innertube Client to use for request
+   */
+  async addVideos(playlist_id: string, video_ids: string[], client?: InnerTubeClient): Promise<{ playlist_id: string; action_result: any }> {
+    throwIfMissing({ playlist_id, video_ids });
+
+    if (!this.#actions.session.logged_in)
+      throw new InnertubeError('You must be signed in to perform this operation.');
+
+    const playlist_edit_endpoint = new NavigationEndpoint({
+      playlistEditEndpoint: {
+        playlistId: playlist_id,
+        actions: video_ids.map((id) => ({
+          action: 'ACTION_ADD_VIDEO',
+          addedVideoId: id
+        }))
+      }
+    });
+
+    const response = await playlist_edit_endpoint.call(this.#actions, {
+      client
+    });
+
+    return {
+      playlist_id,
+      action_result: response.data.actions // TODO: implement actions in the parser
+    };
+  }
+
+  /**
+   * Adds a given playlist to the library of a user.
+   * @param playlist_id - The playlist ID.
+   */
+  async addToLibrary(playlist_id: string){
+    throwIfMissing({ playlist_id });
+
+    if (!this.#actions.session.logged_in)
+      throw new InnertubeError('You must be signed in to perform this operation.');
+
+    const like_playlist_endpoint = new NavigationEndpoint({
+      likeEndpoint: {
+        status: 'LIKE',
+        target: playlist_id
+      }
+    });
+
+    return await like_playlist_endpoint.call(this.#actions, {
+      client: 'TV'
+    });
+  }
+
+  /**
+   * Remove a given playlist to the library of a user.
+   * @param playlist_id - The playlist ID.
+   */
+  async removeFromLibrary(playlist_id: string){
+    throwIfMissing({ playlist_id });
+
+    if (!this.#actions.session.logged_in)
+      throw new InnertubeError('You must be signed in to perform this operation.');
+
+    const remove_like_playlist_endpoint = new NavigationEndpoint({
+      likeEndpoint: {
+        status: 'INDIFFERENT',
+        target: playlist_id
+      }
+    });
+
+    return await remove_like_playlist_endpoint.call(this.#actions, {
+      client: 'TV'
+    });
+  }
```

Comment:
This looks like duplication of code in other existing classes, please don't unnecessarily duplicate code, YouTube.js is already very large as it is.

Reply:
Yeah, at a first glance, I think the problem here is that it needs the client to be `TV`, otherwise it'll fail. Maybe try creating a function that can handle this for all clients and then reuse it (with a `client` parameter).